pylint/lint/run.py

```diff
@@ -231,6 +217,33 @@ def __init__(
             else:
                 sys.exit(self.linter.msg_status)
 
+    def _check_jobs(self) -> None:
+        """Validate jobs config."""
+        if self.linter.config.jobs < 0:
+            # Validate negative jobs
+            print(
+                f"Jobs number ({self.linter.config.jobs}) should be greater than or equal to 0",
+                file=sys.stderr,
+            )
+            sys.exit(32)
+
+        if self.linter.config.jobs != 1:
+            # Validate if jobs is 0 or positive num other than 1
+            if self._plugins:
+                warnings.warn(
+                    "Running pylint in parallel with custom plugins is not currently supported.",
+                    UserWarning,
+                )
+                sys.exit(32)
+            if multiprocessing is None:
```

Comment:
We should move this condition up imo. If multiprocessing is not available then there's ni reason to crash because of it.

Reply:
We don't crash if multiprocessing is None, see line 243...

Reply:
I meant if ``self.linter.config.jobs != 1`` and ``self._plugins`` and ``multiprocessing is None`` are all true, then we're exiting but we could keep linting because the multiprocessing is not going to be activated. We can deactivate the multiprocessing first instead.

```python
if multiprocessing is None:
    print(
        "Multiprocessing library is missing, fallback to single process",
        file=sys.stderr,
    )
    self.linter.set_option("jobs", 1)
if self.linter.config.jobs != 1:
            # Validate if jobs is 0 or positive num other than 1
            if self._plugins:
                warnings.warn(
                    "Running pylint in parallel with custom plugins is not currently supported.",
                    UserWarning,
                )
                sys.exit(32)
elif self.linter.config.jobs == 0:
                self.linter.config.jobs = _cpu_count()
```

Reply:
I can sort of see the argument for your proposal but the point of checking `if multiprocessing is None:` inside `if self.linter.config.jobs != 1:` is to only print `print("Multiprocessing library is missing, fallback to single process",` when the user configured num of jobs is !=1. Moving the check up higher outside of the if statement, means we're going to print that regardless of the number of jobs, even if user passed `-j 1` which seems confusing to me...

anyway, once you read this if you still think it should be changed I will do so.